src/main/java/org/folio/rest/impl/PutOrderLineByIdHelper.java

```diff
@@ -213,10 +214,19 @@ public boolean isUpdateSuccessful() {
             .collect(Collectors.toList());
 
         // Create piece records for item id's which do not have piece records yet
-        itemIds.stream()
-               .filter(id -> !existingItemIds.contains(id))
-               .forEach(itemId -> futuresList.add(createPiece(poLineId, itemId)));
-
+        if(itemIds!=null) {
```

Comment:
Here we are handling a case where partial pieces were created with those associated with item ids. 
We should also handle the case where there were partial pieces created for pieces not associated with item ids.

Reply:
From what I understand, in that case, pieces already exist so we don't need to create again.

Reply:
That is correct.But, Consider the scenario: We are creating 3 pieces without item ids. After 1 piece is created the calls fails. When we retry we are supposed to create the remaining 2 pieces. But that logic is missing here. (The same logic here needs to be rewritten for pieces without itemids)

Reply:
Got it. I have updated the logic. Code coverage went up a bit after this push even without adding a new test which proves we have quite a lot scenarios already covered through unit test :)